models/weight_recorder.cpp

```diff
@@ -155,16 +157,70 @@ nest::weight_recorder::set_status( const DictionaryDatum& d )
     user_set_precise_times_ = true;
 
   device_.set_status( d );
+
+  P_.set( d );
 }
 
+
 void
 nest::weight_recorder::handle( WeightRecorderEvent& e )
 {
   // accept spikes only if detector was active when spike was
   // emitted
   if ( device_.is_active( e.get_stamp() ) )
   {
+
+    // P_sources_ is defined and sender is not in it
+    // or P_targets_ is defined and receiver is not in it
+    if ( ( not P_.sources_.empty()
+           and not std::binary_search(
+                 P_.sources_.begin(), P_.sources_.end(), e.get_sender_gid() ) )
+      or ( not P_.targets_.empty()
+           and not std::binary_search( P_.targets_.begin(),
+                 P_.targets_.end(),
+                 e.get_receiver_gid() ) ) )
+      return;
+
+
     WeightRecorderEvent* event = e.clone();
     B_.events_.push_back( *event );
   }
 }
+
+nest::weight_recorder::Parameters_::Parameters_()
+  : sources_()
+  , targets_()
+{
+}
+
+nest::weight_recorder::Parameters_::Parameters_( const Parameters_& p )
+  : sources_( p.sources_ )
+  , targets_( p.targets_ )
+{
+}
+
+void
+nest::weight_recorder::Parameters_::get( DictionaryDatum& d ) const
+{
+  ( *d )[ nest::names::source ] = sources_;
+  ( *d )[ nest::names::target ] = targets_;
```

Comment:
Since you are inside namespace `nest` here, you do not need the `nest::` qualifiers. Applies also in `set()`.